### ISSUE TITLE
We don't have our own torsocks

### ISSUE THREAD

**Issue by ark3** [telepresenceio/telepresence#576]
Due to #195 as well as #546 and #547, and the fact that the associated Tor Project tickets ([25586](https://trac.torproject.org/projects/tor/ticket/25586) and [25627](https://trac.torproject.org/projects/tor/ticket/25627)) haven't seen any movement and probably won't for an unclear time period, we should distribute our own build of torsocks. It should live in its own package since it will not change as often as Telepresence. This is true for our sshuttle as well, as in #533. Once we get this done, we can close #528 too.

**Update:** See [checklist](https://github.com/telepresenceio/telepresence/issues/576#issuecomment-492274837) below.

**Comment by ark3** [comment]
FWIW the fixes for those Tor Project tickets appear to have been landed. However, there is no release in sight at the moment.

**Comment by alnr** [comment]
I think the fix has landed in torsocks v2.3.0: https://gitweb.torproject.org/torsocks.git
This could presumably fix #195 #423 #546 #547 ?
Let me know if I can help.

**Comment by ark3** [comment]
Can you get it set up on Linux and MacOS (it doesn't seem to be in Homebrew yet)? If so, please try [this sequence](https://gist.github.com/ark3/95ea11bec7f368d8e2a73a7e2f383f34). Let's see whether stuff really works! 😉

**Comment by alnr** [comment]
Using a [patched](https://trac.torproject.org/projects/tor/ticket/28538) torsocks 2.3.0 library (macOS): [result](https://gist.github.com/alnr/8c6f3c17dd8e3d4e5f5914599b16fa09)
Is that the expected result?

**Comment by ark3** [comment]
The second request (10.0.0.1) was intended to yield an error. It looks like your cluster gives back a result for that. Can you try an IP that will yield an error? Thanks!

**Comment by alnr** [comment]
```
>>> socket.gethostbyaddr("123.123.123.123")
('123.123.123.123', [], ['123.123.123.123'])
```

**Comment by alnr** [comment]
I have gathered more data points on the general issue of `-m inject-tcp`, and I believe everything is fixed by `torsocks-2.3.0`.

My team uses `-m inject-tcp` successfully on both Linux and with NodeJS and it works beautifully.

Let me know if I can help this issue along. There seem to be many many tickets related to the `inject-tcp` method, most of them I think are fixed by the torsocks update. On macOS, I have updated torsocks in homebrew to 2.3.0 (https://github.com/Homebrew/homebrew-core/pull/39501).
